src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -391,6 +399,26 @@ ss::future<> metadata_dissemination_service::update_leaders_with_health_report(
     }
 }
 
+namespace {
+std::vector<cluster::ntp_leader> from_ntp_leader_revision_vector(
+  std::vector<cluster::ntp_leader_revision> leaders) {
+    std::vector<cluster::ntp_leader> old_leaders;
+    old_leaders.reserve(leaders.size());
+    std::transform(
+      leaders.begin(),
+      leaders.end(),
+      std::back_inserter(old_leaders),
+      [](cluster::ntp_leader_revision& leader) {
+          return cluster::ntp_leader{
```

Comment:
Should this be initializing revision to an invalid value?  I don't think the ntp_leader class has a default value for the attribute

Reply:
`ntp_leader` has no `revision_id` field. Whenever signed named type is used it is always initialized to minimum possible value.

Reply:
Sorry, I was reading this function backwards, thought it was converting in the other direction to `ntp_leader_revision`